### PR TITLE
Fixing global.db upgrade for older versions

### PR DESCRIPTION
**Pull request by pereyra-m**
|Related issue|
|---|
|#6216|

## Description

Older versions of **global.db** database didn't have columns like _register_ip_ (introduced in 3.9) or had different data types (**last_keepalive** as TEXT instead of INTEGER before 3.10). The  upgrade mechanism now also checks for manager's last keepalive because it was the last change before adding the metadata table.

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
  - [x] Source installation
- [ ] Package installation
- [x] Source upgrade
- [ ] Package upgrade
- [ ] Review logs syntax and correct language
- [ ] QA templates contemplate the added capabilities

- [x] Retrocompatibility with older Wazuh versions

### CODE REVIEW

**Review by DProvinciani — APPROVED** [review]
Considering the @vikman90 comment telling that the white lines will be removed in a separate PR. Approving this PR.

**Review by palaciosjeremias — APPROVED** [review]
LGTM

File: src/wazuh_db/wdb_global.c [hunk 1]
```diff
@@ -1383,3 +1383,24 @@ wdbc_result wdb_global_get_all_agents(wdb_t *wdb, int* last_agent_id, char **out
...
+// Check the agent 0 status in the global database
+int wdb_global_check_manager_keepalive(wdb_t *wdb) {
+    if (wdb_stmt_cache(wdb, WDB_STMT_GLOBAL_CHECK_MANAGER_KEEPALIVE) < 0) {
+        merror("DB(%s) Can't cache statement", wdb->id);
+        return -1;
+    }
+
+    sqlite3_stmt *stmt = wdb->stmt[WDB_STMT_GLOBAL_CHECK_MANAGER_KEEPALIVE];
+
+    switch (sqlite3_step(stmt)) {
```
Comment (pereyra-m) [inline src/wazuh_db/wdb_global.c:1396]: Would it be necessary to finalize this statement?

File: src/wazuh_db/wdb.h [hunk 2]
```diff
@@ -1706,4 +1707,17 @@ wdbc_result wdb_global_get_all_agents(wdb_t *wdb, int* last_agent_id, char **out
 // Finalize a statement securely
 #define wdb_finalize(x) { if (x) { sqlite3_finalize(x); x = NULL; } }
 
+/**
+ * @brief Check the agent 0 status in the global database
+ *
+ * The table "agent" must have a tuple with id=0 and last_keepalive=1999/12/31 23:59:59 UTC.
```
Comment (pereyra-m) [inline src/wazuh_db/wdb.h:1713]: Typo: the year 1999

**Review by pereyra-m — COMMENTED** [review]
LGTM!

**Pull request merged**